contributingGuides/STYLE.md

```diff
@@ -477,20 +477,63 @@ if (ref.current && 'getBoundingClientRect' in ref.current) {
 
 ### Default value for inexistent IDs
 
- Use `'-1'` or `-1` when there is a possibility that the ID property of an Onyx value could be `null` or `undefined`.
+Use `CONST.DEFAULT_NUMBER_ID` when there is a possibility that the number ID property of an Onyx value could be `null` or `undefined`. **Do not default string IDs to any value unless absolutely necessary**, in case it's necessary use `CONST.DEFAULT_STRING_ID` instead.
```

Comment:
```suggestion
Use `CONST.DEFAULT_NUMBER_ID` when there is a possibility that and ID that's a number could be `null` or `undefined`. 

Use `CONST.DEFAULT_STRING_ID` when there is a possibility that and ID that's a string could be `null` or `undefined`. 

**Do not default IDs to any value unless absolutely necessary**
```

Reply:
I'm confused by that suggestion.

Can we please include some examples for why we have these guidelines and our reasoning for them? I think explaining the why will help people follow the suggestions.

Reply:
> unless absolutely necessary

Please also give an example of when this is necessary

Reply:
@neil-marcellini I've added some explanations as requested, please let me know if it's good enough

> Please also explain and give an example of when this is necessary

@iwiznia Unfortunately I couldn't think of any examples default the string ID would be necessary. We just added because, given the codebase's size, it *could* happen so we are already covering this situation in the guidelines. If you think this is not valid we can instead remove this exception and `DEFAULT_STRING_ID` for now.

Reply:
> Didn't we say in the thread you linked that we needed to default to `undefined` or `0` when connecting to an onyx key?

That's true, but that would be not providing a default, or using a number right?

It's sounding like we really can't come up with a good case for defaulting a string. I think it would be more clear to leave that out entirely. We could add a note that if you find a reasonable exception or are confused about a case to raise your question in #expensify-open-source.

Reply:
I removed `DEFAULT_STRING_ID` and added the note.